File: Raven.Client.Java/src/main/java/net/ravendb/abstractions/commands/ICommandData.java
package net.ravendb.abstractions.commands;

import net.ravendb.abstractions.data.Etag;
import net.ravendb.abstractions.data.HttpMethods;
import net.ravendb.abstractions.json.linq.RavenJObject;

/**
 * A single operation inside the batch.
 *
 */
public interface ICommandData {

  /**
   * Gets the key
   * @return
   */
  public String getKey();

  /**
   * Gets the method
   * @return
   */
  public HttpMethods getMethod();

  /**
   * Gets the Etag.
   * @return
   */
  public Etag getEtag();

  /**
   * Gets the metadata.
   * @return
   */
  public RavenJObject getMetadata();

  /**
   * Gets the additional data.
   * @return
   */
  public RavenJObject getAdditionalData();

  /**
   * Sets the additional metadata
   * @param additionalMeta
   */
  public void setAdditionalData(RavenJObject additionalMeta);

  /**
   * Translate the instance to a Json object.
   * @return
   */
  public RavenJObject toJson();

}

File: src/Raven.NewClient/Commands/ICommandData.cs
//-----------------------------------------------------------------------
// <copyright file="ICommandData.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using Raven.Abstractions.Data;
using Raven.NewClient.Json.Linq;

namespace Raven.Abstractions.Commands
{
    /// <summary>
    /// A single operation inside a batch
    /// </summary>
    public interface ICommandData
    {
        /// <summary>
        /// Key of a document.
        /// </summary>
        string Id { get; }

        /// <summary>
        /// Returns operation method.
        /// </summary>
        string Method { get; }

        /// <summary>
        /// Current document etag, used for concurrency checks (null to skip check)
        /// </summary>
        long? Etag { get; }

        /// <summary>
        /// Additional command data. For internal use only.
        /// </summary>
        RavenJObject AdditionalData { get; set; }

        /// <summary>
        /// Translates this instance to a Json object.
        /// </summary>
        /// <returns>RavenJObject representing the command.</returns>
        RavenJObject ToJson();
    }
}
